src/utils/mask-object.ts

```diff
@@ -9,7 +9,7 @@ export const maskObject = (object: any, options?: MaskOptions) => {
   if (!object || typeof object !== 'object') return object;
   if (!options?.maskProperties && !options?.hideProperties) return object;
 
-  const copyObject = { ...object };
+  const copyObject = JSON.parse(JSON.stringify(object));
```

Comment:
This could result in error in some edge cases

Reply:
As it is used exclusively as a util under logger context, I think we'd be OK here as the logger objects are likely to be serializable otherwise the logger itself would error. Unless I'm missing something?